libpod/container_internal_common.go

```diff
@@ -568,6 +568,11 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
 	g.SetRootPath(c.state.Mountpoint)
 	g.AddAnnotation("org.opencontainers.image.stopSignal", strconv.FormatUint(uint64(c.config.StopSignal), 10))
 
+	if c.config.StopTimeout != 0 {
+		annotation := fmt.Sprintf("uint64 %d", c.config.StopTimeout*1000000) // sec to usec
+		g.AddAnnotation("org.systemd.property.TimeoutStopUSec", annotation)
```

Comment:
So I assume the oci runtime sets this on the slice then? Does work with both crun and runc?
Also in this case do we also have to set the stop signal we use or is this already done?

Reply:
it is on the scope.  I'll fix the git commit message.  It works both with crun and runc, and if an OCI runtime doesn't support it then it is just ignored.

I don't see a way to override the stop signal using the same mechanism

Reply:
Ok I took a quick look and testing it we do not pass down the proper stop signal to the scope which means if the container doesn't react on SIGTERM (and uses a different signal) the fix will not help.

This should do it, mind adding it here as well and add it to the test?
```diff
diff --git a/libpod/container_internal_common.go b/libpod/container_internal_common.go
index 1fdca63cf..72564254a 100644
--- a/libpod/container_internal_common.go
+++ b/libpod/container_internal_common.go
@@ -573,6 +573,10 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
                g.AddAnnotation("org.systemd.property.TimeoutStopUSec", annotation)
        }
 
+       if c.config.StopSignal != 0 {
+               g.AddAnnotation("org.systemd.property.KillSignal", strconv.FormatUint(uint64(c.config.StopSignal), 10))
+       }
+
        if _, exists := g.Config.Annotations[annotations.ContainerManager]; !exists {
                g.AddAnnotation(annotations.ContainerManager, annotations.ContainerManagerLibpod)
        }
```

Reply:
thanks!  I've amended the change and the test